packages/dialog/src/react/index.js

```diff
@@ -78,6 +80,7 @@ class ModalOverlay extends React.Component {
 }
 
 ModalOverlay.propTypes = {
+  'aria-label': PropTypes.string.isRequired,
```

Comment:
so I think I like using the actual aria attrs and providing prop warnings. I think it helps teach a little accessibility along the way. the tricky part is there are sometimes multiple ways to do the right think. in this case we could also provide prop help(or docs) on how to do `aria-labelledby` or `aria-description` instead of the `aria-label`. we should explore good patterns that are flexible.

I also agree that sometimes a prop might just do more than it's native features. i could see that being a different prop in conjuction with the aria prop - and maybe prop warning(or docs) to expose what is expected and what is happening. in this pr the `aria-label` is only required if the `modal` prop is true - for example.